### PR TITLE
improvement: allow mutagen code sync outside subpath

### PR DESCRIPTION
**Pull request by Orzelius**
allow mutagen code sync outside subpath and add dev-mode to the [build dependencies example](https://github.com/garden-io/garden/tree/sync-build-deps/examples/build-dependencies)

### CODE REVIEW

File: core/src/plugins/container/config.ts [hunk 1]
```diff
@@ -253,6 +253,17 @@ export const syncDefaultGroupSchema = () =>
...
   hotReloadSyncSchema().keys({
     exclude: syncExcludeSchema(),
+    source: joi // same as hotReloadSyncSchema but no subPathOnly
+      .posixPath()
+      .relativeOnly()
+      .allowGlobs()
+      .default(".")
+      .description(
+        deline`
+      POSIX-style path of the directory to sync to the target, relative to the module's top-level directory.
```
Comment (vvagaytsev) [inline core/src/plugins/container/config.ts:263]: Does it still have to be a relative path?

Comment (edvald) [reply core/src/plugins/container/config.ts:263]: I think we should allow a full path actually, because absolute paths may be templated in.

Comment (Orzelius) [reply core/src/plugins/container/config.ts]: makes sense, allowed in [0aa7cc0](https://github.com/garden-io/garden/pull/3145/commits/0aa7cc05c54854322e396d95825b5784aaf2662f)

**Review by vvagaytsev — APPROVED** [review]

**Review by eysi09 — APPROVED** [review]
LGTM.

**Pull request merged**